api/controllers/auth.js

```diff
@@ -77,18 +81,27 @@ const resetApprovementPassword = async (req, res, next) => {
       });
     }
   } catch (e) {
-    next(e);
+    throw new error.ResetPasswordApproveError(e.message);
   }
 };
 
 const resetPassword = async (req, res, next) => {
   const { linkId, password, passwordConfirm } = req.body;
-  try {
-    const v = await passwordSchema.validate({ password, passwordConfirm });
-    const a = v && (await services.resetPassword({ password, linkId }));
-    res.status(200).send(a);
-  } catch (e) {
-    next(e);
+  if (!linkId || !password || !passwordConfirm) {
+    throw new errors.ResetPasswordError('Empty params');
+  }
+  if (linkId && password && passwordConfirm) {
+    try {
```

Comment:
It is not needed to use both of this condition.

Reply:
Done